base/htime.h

```diff
@@ -72,8 +72,11 @@ static inline unsigned long long timestamp_ms() {
 void msleep(unsigned int ms);
 // ms
 unsigned int gettick();
+
+#ifndef OS_SOLARIS  // Solaris has built-in gethrtime().
```

Comment:
Solaris has built-in `gethrtime()` in <sys/time.h> for high resolution time support, and also returns nanoseconds. Use that instead.